packages/ember-application/lib/system/engine.js

```diff
@@ -0,0 +1,422 @@
+/**
+@module ember
+@submodule ember-application
+*/
+import Namespace from 'ember-runtime/system/namespace';
+import Registry from 'container/registry';
+import RegistryProxy from 'ember-runtime/mixins/registry_proxy';
+import DAG from 'dag-map';
+import { get } from 'ember-metal/property_get';
+import { set } from 'ember-metal/property_set';
+import { assert, deprecate } from 'ember-metal/debug';
+import { canInvoke } from 'ember-metal/utils';
+import EmptyObject from 'ember-metal/empty_object';
+import DefaultResolver from 'ember-application/system/resolver';
+import EngineInstance from './engine-instance';
+
+function props(obj) {
+  var properties = [];
+
+  for (var key in obj) {
+    properties.push(key);
+  }
+
+  return properties;
+}
+
+/**
```

Comment:
Can you add a small description here?  This is the header page that shows up for the class in our API docs...